cli/azd/internal/appdetect/java.go

```diff
@@ -381,3 +394,41 @@ func contains(array []string, str string) bool {
 	}
 	return false
 }
+
+func parseProperties(properties Properties) map[string]string {
+	result := make(map[string]string)
+	for _, entry := range properties.Entries {
+		result[entry.XMLName.Local] = entry.Value
+	}
+	return result
+}
+
+func detectSpringBootVersion(currentRoot *mavenProject, mavenProject *mavenProject) string {
+	if currentRoot != nil {
+		return detectSpringBootVersionFromProject(currentRoot)
+	} else if mavenProject != nil {
+		return detectSpringBootVersionFromProject(mavenProject)
```

Comment:
It's possible that mavenProject and current Root use different version. In this case, mavenProject should have higher priority.

Reply:
Updated.